ios/RNSScreenStackHeaderSubview.mm

```diff
@@ -72,6 +72,7 @@ - (void)updateLayoutMetrics:(const react::LayoutMetrics &)layoutMetrics
         self);
   } else {
     self.bounds = CGRect{CGPointZero, frame.size};
+    [self.superview layoutIfNeeded];
```

Comment:
Okay, so we now trigger layout in the native superview (this is one of system views, beyond control of react) every time we receive a new frame from react. 

The solution itself is ok & if it helps I'm for it, but do we have understanding why does it not work without this change? Especially that we do not have such code on Paper & I believe it works fine there?

Reply:
Did you check what happens if you disable view recycling on header subviews in this PR? Im speaking with context of this issue on which we debated earlier today: https://github.com/software-mansion/react-native-screens/issues/2232

Reply:
This might be the same issue of the views being swapped due to recycling pool working in stack manner.

Reply:
Turning off recycling does not fix this issue